app/ui/legacy/src/main/java/com/fsck/k9/fragment/MessageListFragment.java

```diff
@@ -1585,6 +1586,29 @@ private void copyOrMove(List<MessageReference> messages, long destinationFolderI
         }
     }
 
+    private void saveToDraftAndRemove(List<MessageReference> messages){
+
+        List<MessageReference> draftsSaved = new ArrayList<>(selected.size());
+        for(MessageReference messageReference: messages) {
```

Comment:
This code should…
a) live in `MessagingController`
b) run in a background thread because it's reading from and writing to disk (`MessagingController` provides infrastructure for that, see `putBackground()`)